modules/dnn/test/test_layers_1d.cpp

```diff
@@ -245,4 +245,39 @@ INSTANTIATE_TEST_CASE_P(/*nothing*/, Layer_Elemwise_1d_Test, Combine(
 /*operation*/           Values("div", "prod", "max", "min", "sum")
 ));
 
+// WORKING FOR 1D
+typedef testing::TestWithParam<tuple<int, std::vector<int>>> Layer_Permute_Test;
+TEST_P(Layer_Permute_Test, Accuracy)
+{
+    LayerParams lp;
+    lp.type = "Permute";
+    lp.name = "PermuteLayer";
+
+    int order[] = {0}; // Since it's a 0D tensor, the order remains [0]
+    lp.set("order", DictValue::arrayInt(order, 1));
+    Ptr<PermuteLayer> layer = PermuteLayer::create(lp);
+
+    int dims = get<0>(GetParam());
+    std::vector<int> input_shape = get<1>(GetParam());
+
+
+    Mat input = Mat(dims, input_shape.data(), CV_32F);
+    cv::randn(input, 0.0, 1.0);
+    Mat output_ref = Mat(dims, input_shape.data(), CV_32F, input.data);
+
+    std::vector<Mat> inputs{input};
+    std::vector<Mat> outputs;
+
+    runLayer(layer, inputs, outputs);
+    ASSERT_EQ(shape(output_ref), shape(outputs[0]));
+    normAssert(output_ref, outputs[0]);
+}
+INSTANTIATE_TEST_CASE_P(/*nothing*/,  Layer_Permute_Test,
+/*input blob shape*/ testing::Values(
+            std::make_tuple(0, std::vector<int>{0}),
```

Comment:
Not a question for this PR, but in general. As I see, `Mat(0, {0}, CV_32F);` can be initialized by `randn` while `Mat(0, {}, CV_32F);` not?

Reply:
In other words, shall we consider all the following initializations as scalar Mat?
```
Mat(0, {0}, CV_32F);
Mat(1, {0}, CV_32F);
Mat(0, {}, CV_32F);
```

Reply:
I guess only this one is valid


`Mat(0, {0}, CV_32F);` where `dims = 0` and shape is `{0}`